graveler/committed/tree/parts_and_values.go

```diff
@@ -0,0 +1,78 @@
+package tree
+
+import "github.com/treeverse/lakefs/graveler"
+
+type partsAndValuesIterator struct {
+	repo  Repo
+	parts []Part
+	it    graveler.ValueIterator
+	err   error
+}
```

Comment:
This iterator also behaves a bit differently as the expected behavior from our current Iterators / Scanners are to call Next, on true you can access Value to get the current iterator value.